pyiron/dft/job/generic.py

```diff
@@ -270,6 +270,8 @@ def set_kpoints(
             if mesh is not None:
                 warnings.warn("mesh value is overwritten by k_mesh_spacing")
             mesh = self.get_k_mesh_by_cell(k_mesh_spacing=k_mesh_spacing)
+        elif self.k_mesh_spacing is not None: # switching off k_mesh_spacing would preserve the current mesh
+            mesh = self.get_k_mesh_by_cell(k_mesh_spacing=self.k_mesh_spacing)
```

Comment:
This would overwrite `mesh` regardless of what was set, so you have to make sure that `mesh` is not `None`